dplython/dplython.py

```diff
@@ -72,8 +72,16 @@ def ungroup(self):
     self._grouped_on = None
     self._grouped_self = None
 
+  def regroup(self, names):
+    self.group_self(names)
+    return self
+
   def apply_on_groups(self, delayedFcn):
-    if isinstance(delayedFcn, mutate) or isinstance(delayedFcn, sift):
+    # use names and a list instead of isinstance here to keep things manageable?
+    # if everything goes to verbs, can we even remove the apply on groups altogether?
+    if isinstance(delayedFcn, mutate) or isinstance(delayedFcn, sift) or isinstance(delayedFcn, inner_join) or \
+            isinstance(delayedFcn, full_join) or isinstance(delayedFcn, left_join) or \
+            isinstance(delayedFcn, right_join):
```

Comment:
Like you say in the comment above, this would be cleaner with:
```
handled_classes = (mutate, sift, inner_join, full_join, left_join, right_join)
if isinstance(delayedFcn, handled_classes):
  return delayedFcn(self)
```
Note that `handled_classes` has to be a tuple, not a list.